### PR TITLE
[opentitanlib] Move verilator transport to SerialPortUart

### PR DESCRIPTION
**Pull request by jwnrt**
This PR switches the verilator transport from its custom `Uart` implementation to the `SerialPortUart` that HyperDebug and ChipWhisperer transports use, but at a lower baud.

Note on differences between `VerilatorUart` and `SerialPortUart`:

1. `VerilatorUart` ignores baud rate setting. It's backed by a `pty`, so this is already ignored.
2. `VerilatorUart` does not create a lock file, so we gain that.
3. `VerilatorUart` doesn't have the "pacing" on writes (a delay after every byte is sent).
   * This shouldn't cause errors for Verilator. I'm somewhat surprised this delay is needed, though.

### CODE REVIEW

**Review by nbdd0121 — APPROVED** [review]

File: sw/host/opentitanlib/src/transport/chip_whisperer/mod.rs [hunk 1]
```diff
@@ -27,6 +27,8 @@ pub mod gpio;
 pub mod spi;
 pub mod usb;
 
+const UART_BAUD: u32 = 115200;
```
Comment (engdoreis) [inline sw/host/opentitanlib/src/transport/chip_whisperer/mod.rs:30]: Can this be on `board.rs`?

**Review by engdoreis — COMMENTED** [review]
Thanks for the PR. 
I think that the baud should come from the json file. See example:
https://github.com/lowRISC/opentitan/blob/78dee87b4ab0047d3815ae76a0012dbc49e9a9ed/sw/host/opentitanlib/src/app/config/opentitan.json#L136C18-L136C18

File: sw/host/opentitanlib/src/transport/chip_whisperer/mod.rs (hunk 1 above)
Comment (jwnrt) [reply sw/host/opentitanlib/src/transport/chip_whisperer/mod.rs]: Okay, moved

**Comment by jwnrt** [comment]
I agree this should be loaded from the `.json` files, but I'm not sure how to propagate that from the config to where the UART is created.

Would you mind that being left for a future PR, or should it be implemented here?

**Comment by engdoreis** [comment]
> I agree this should be loaded from the `.json` files, but I'm not sure how to propagate that from the config to where the UART is created.
> 
> Would you mind that being left for a future PR, or should it be implemented here?

Yes, could you open an issue?

**Review by engdoreis — APPROVED** [review]

**Pull request merged**